tests/flow_counter/flow_counter_utils.py

```diff
@@ -0,0 +1,278 @@
+            set_route_flow_counter_pattern(self.dut, route_pattern)
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        """Do following tasks:
+            1. Verify route flow counter stats agaist expected value
+            2. Disable route flow coutern and remove route pattern
+
+        Args:
+            exc_type (object): not used
+            exc_val (object): not used
+            exc_tb (object): not used
+        """
+        if not is_route_flow_counter_supported(self.dut):
+            return
+
+        try:
+            result, message = self.check_stats()
+            pytest_assert(result, message)
+        finally:
+            set_route_flow_counter_status(self.dut, False)
+            for route_pattern in self.route_pattern_list:
+                remove_route_flow_counter_pattern(self.dut, route_pattern)
+
+
+    def check_stats(self):
+        """Verify route flow counter statistic
+
+        Returns:
+            tuple: (status, error message)
+        """
+        logger.info('Checking route flow counter stats')
+        actual_stats = parse_route_flow_counter_stats(self.dut)
+        result, message = verify_route_flow_counter_stats(self.expected_stats, actual_stats)
+        if not result:
+            return result, message
+
+        if len(self.expected_stats) > 0:
+            logger.info('Checking route flow counter stats after clearing by route')
+            to_clear = random.sample(list(self.expected_stats.keys()), 1)[0]
+            clear_route_flow_counter_by_route(self.dut, to_clear)
+            for key in self.expected_stats[to_clear]:
+                self.expected_stats[to_clear][key] = '0'
+            actual_stats = parse_route_flow_counter_stats(self.dut)
+            result, message = verify_route_flow_counter_stats(self.expected_stats, actual_stats)
+            if not result:
+                return result, message
+
+        if len(self.expected_stats) == 1 and len(self.route_pattern_list) == 1:
+            logger.info('Checking route flow counter stats after clearing by pattern')
+            clear_route_flow_counter_by_pattern(self.dut, self.route_pattern_list[0])
+        else:
+            logger.info('Checking route flow counter stats after clearing all routes')
+            clear_route_flow_counter(self.dut)
+        for prefix, value in self.expected_stats.items():
+            for key in value:
+                self.expected_stats[prefix][key] = '0'
+
+        actual_stats = parse_route_flow_counter_stats(self.dut)
+        return verify_route_flow_counter_stats(self.expected_stats, actual_stats)
+
+
+def is_route_flow_counter_supported(dut):
+    """Check if route flow counter is supported on this platform
+
+    Args:
+        dut (object): DUT object
+
+    Returns:
+        bool: True if supported
+    """
+    skip, _ = check_skip_release(dut, skip_versions)
+    if skip:
+        return False
+
+    global support_route_flow_counter
```

Comment:
I would prefer to keep this because:
1. is_route_flow_counter_supported is used by RouteFlowCounterTestContext, which is not a test function. We can't call fixture directly, and we have to pass is_route_flow_counter_supported to RouteFlowCounterTestContext if it is a fixture. 
2. even if we turn is_route_flow_counter_supported into a fixture, the global variable is still needed because we are using `wait_until` to get the capability, wait_until cannot return value (an option is to use a list as output parameter of get_route_flow_counter_capability which stores the capability, but it does not simply the code from my POV).

Reply:
I believe that passing a list/dict will make it more readable and safe

Reply:
ok, will fix